doc/deploying_gcs_on_bare_metal_machines.md

```diff
@@ -0,0 +1,196 @@
+1. Create a yaml file of the format presented below.<br />
+   ```
+   ---
+   kind: PersistentVolumeClaim
+   apiVersion: v1
+   metadata:
+      name: gcs-pvc
+   spec:
+      storageClassName: glusterfs-csi
+      accessModes:
+         - ReadWriteMany
+      resources:
+        requests:
+           storage: 2Gi
+   ```
+   `gcs-pvc` is name of the pvc that is to be created, and the size of volume is `2Gi`.
+   Name of file : pvc.yaml
+
+2. To create PVC, run the following command.<br />
+   `$ kubectl create -f pvc.yaml`
+
+3. To check whether the PVC is created or not run the following command.
+   `$ kubectl get pvc`
+
+   If the PVC is in BOUND state then the volume is created and can be used now.
+
+#### Creating parallel PVCs:
+
+To create parallel PVCs, run the "pvc create" command in loop without waiting for the previous PVC to be in BOUND state. You will have to change the name of pvc in the pvc.yaml file to create a new pvc.<br />
+
+To create parallel PVC using python:<br />
+```
+import re
+import os
+for i in range(1, 500):
+    # Changing name of PVC before creating. 
+    fp = open(r'pvc.yaml', 'r')
+    new_name = 'gcs-pvc' + str(i)
+    replace_data = fp.read().replace('gcs-pvc', new_name)
+    fp.seek(0,0)
+    fw = open(r'pvc.yaml', 'w')
+    fw.write(replace_data)
+    fw.seek(0,0)
+    # Create PVC
+    os.system('kubectl create -f pvc.yaml')
+    # Renaming PVC back to original name
+    fp = open(r'pvc.yaml', 'r')
+    replace_data = fp.read().replace(new_name, 'gcs-pvc')
+    fp.seek(0, 0)
+    fw = open(r'pvc.yaml', 'w')
+    fw.write(replace_data)
+    fw.seek(0, 0)
+```
+
+### Scale Testing results:
+
+We deployed GCS by following the steps mentioned above. We created 1000 volumes in batch of 250, and deleted 1000 volumes in the batch of 10. The scale testing results can be found in this [speadsheet](https://docs.google.com/spreadsheets/d/1nqySz3R2uR7MUPWWxzJMVWETMob-Mkwh67HmP6QHPDQ/edit?usp=sharing).
+
+### Remove GCS setup
+
+To remove gcs setup, run:<br />
+`$ kubectl delete namespace gcs`
+
+> Note : wait for the command to complete.
+
+### Remove kubernetes cluster
+
+To remove the kubernetes setup, run:<br />
+`$ ansible-playbook -i <path to inventory file>/<name of inventory file>  <path to deploy directory>/deploy/kubespray/reset.yml`
+
+### Troubleshooting
+
+* If deploying kubernetes fails with error: `Failure: "No package docker-ce available.` <br />
+  Then one must manually install docker-ce on the system.
+  ##### In Centos:<br />
+  `$ yum install docker` <br />
+  `$ systemctl status docker  #To check whether docker is runnning or not` <br />
+  ##### In Fedora
+  ```
+  $ sudo dnf -y install dnf-plugins-core
+  $ sudo dnf config-manager --add-repo https://download.docker.com/linux/fedora/docker-ce.repo
+  $ sudo dnf config-manager --set-enabled docker-ce-edge
+  $ sudo dnf config-manager --set-enabled docker-ce-test
+  $ sudo dnf install docker-ce
+  $ sudo systemctl start docker
+  ```
```

Comment:
the steps to trouble shoot when the pods don't change to the "ready" state is missing.

Reply:
I think for this I will have to cover a wide range of issues depending on the pod that failed. This can be added later.